src/App.tsx

```diff
@@ -1,27 +1,83 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum Sort {
+  All = 'all-button',
+  FirstFive = 'first-five-button',
+  Red = 'red-button',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const getButtonName = (sortType: Sort) => {
+  switch (sortType) {
+    case Sort.All:
+      return 'Load all goods';
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    case Sort.FirstFive:
+      return 'Load 5 first goods';
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    case Sort.Red:
+      return 'Load red goods';
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    default:
+      return '';
+  }
+};
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sort, setSort] = useState<Sort>();
+
+  const getGoods = useCallback(async (promise: Promise<Good[]>) => {
+    setGoods(await promise);
+  }, []);
+
+  const handleClick = useCallback((sortType: Sort) => {
+    if (sort === sortType) {
+      return;
+    }
+
+    switch (sortType) {
+      case Sort.All:
+        setSort(Sort.All);
+        getGoods(getAll());
+        break;
+
+      case Sort.FirstFive:
+        setSort(Sort.All);
+        getGoods(get5First());
+        break;
+
+      case Sort.Red:
+        setSort(Sort.Red);
+        getGoods(getRedGoods());
+        break;
+
+      default:
+        break;
+    }
```

Comment:
```suggestion
    }
    
    setSort(sortType);
```

Instead of repeating it multiple times